### ISSUE TITLE
getContent always returns something

### ISSUE THREAD

**Issue by nlemoine** [bolt/bolt#6938]
Using the `query` service provider, the `getContent` always returns a result when it shouldn't. 

Details
-------

 - Relevant Bolt Version: 3.3.2
 - Install type: Composer install
 - PHP version: 7.1.6
 - Used web server: Apache 2.4

Reproduction
------------

 1. **Bug summary**: 

Tried to query some content type with a where clause on an `id` parameter. When the `id` parameter is `null` or `false`, the query still returns something. 

  2. **Steps to reproduce**:

```php
$this->app['query']->getContent('pages', [
    'id'           => false,
])
```
 3. **Expected result**: 

No results 

 4. **Actual result**:

Always returns some "fallback" results

**Comment by GawainLynch** [comment]
I'll let @rossriley confirm, but my understanding of the intent of design is to match Doctrine, and always return a result set, empty or otherwise, unless it is a single record request.

**Comment by nlemoine** [comment]
This behaviour still applies on: 

```php
$this->app['query']->getContent('pages', [
    'id'           => false,
    'returnsingle' => true,
]);
```

**Comment by GawainLynch** [comment]
Fixed in #7110
